### PR TITLE
Feature/dp 229 be share my information generating code api

### PR DESCRIPTION
**Pull request by maciej-goaco**

### CODE REVIEW

File: Services/CO.CDP.DataSharing.WebApi.Tests/UseCase/GenerateShareCodeUseCaseTest.cs [hunk 1]
```diff
@@ -5,53 +5,130 @@
 using CO.CDP.OrganisationInformation.Persistence.Forms;
 using FluentAssertions;
 using Moq;
+using CO.CDP.Authentication;
```
Comment (glgoaco) [inline Services/CO.CDP.DataSharing.WebApi.Tests/UseCase/GenerateShareCodeUseCaseTest.cs:8]: Consider using `EntityFactory.GivenOrganisation` to reduce repetitive setup code and improve test maintainability

File: Services/CO.CDP.DataSharing.WebApi/UseCase/GenerateShareCodeUseCase.cs [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
+    IFormRepository formRepository,
+    IMapper mapper)
     : IUseCase<ShareRequest, ShareReceipt>
 {
     public async Task<ShareReceipt> Execute(ShareRequest shareRequest)
     {
+        var org = await organisationRepository.Find(shareRequest.OrganisationId);
+        if (org == null || claimService.GetOrganisationId() != org.Id)
+        {
+            throw new InvalidOrganisationRequestedException("Invalid Organisation requested.");
```
Comment (glgoaco) [inline Services/CO.CDP.DataSharing.WebApi/UseCase/GenerateShareCodeUseCase.cs:22]: Is there a plan to unique share codes?

Comment (maciej-goaco) [reply Services/CO.CDP.DataSharing.WebApi/UseCase/GenerateShareCodeUseCase.cs:22]: Discussed offline with Gavin, part of previous PR, agreed to mark as resolved.

File: Services/CO.CDP.DataSharing.WebApi.Tests/UseCase/GenerateShareCodeUseCaseTest.cs (hunk 1 above)
Comment (maciej-goaco) [reply Services/CO.CDP.DataSharing.WebApi.Tests/UseCase/GenerateShareCodeUseCaseTest.cs:8]: done.

File: Services/CO.CDP.DataSharing.WebApi/appsettings.json [hunk 3]
```diff
@@ -7,6 +7,7 @@
   },
   "AllowedHosts": "*",
   "Version": "dev",
+  "Organisation:Authority": "",
```
[the remark this answers is not in the document]
Comment (maciej-goaco) [reply Services/CO.CDP.DataSharing.WebApi/appsettings.json:10]: done.

**Pull request merged**